src/utilities/urlAssembly.ts

```diff
@@ -1,10 +1,11 @@
 import { Param, Resolved, Route } from '@/types'
 import { setParamValuesOnUrl } from '@/utilities/paramsFinder'
 
-export function assembleUrl(route: Resolved<Route>, values: Record<string, unknown> = {}): string {
+export function assembleUrl(route: Resolved<Route>, values: Record<string, unknown[]> = {}): string {
   const params = Object.entries<Param[]>(route.params)
+  const pathWithQuery = route.query.length ? `${route.path}?${route.query}` : route.path
 
   return params.reduce((url, [name, params]) => {
     return setParamValuesOnUrl(url, { name, params, values: values[name] })
```

Comment:
Super cool you didn't actually have to write any additional logic for filling in the params

Reply:
yeah, kudos to you for finding the syntax that just works and feels good